Packages/com.unity.inputsystem/InputSystem/Actions/Interactions/PressInteraction.cs

```diff
@@ -93,12 +93,13 @@ public void Process(ref InputInteractionContext context)
                             if (isActuated)
                                 context.PerformedAndStayPerformed();
                             else
-                                context.Cancelled();
+                            {
+                                context.PerformedAndGoBackToWaiting();
+                                context.Waiting();
```

Comment:
separate `Waiting()` call should not be necessary. `PerformedAndGoBackToWaiting()` triggers a "current -> Peformed -> Waiting" state change cycle.